File: zemiao/283.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i,j = 0,0
        while j<len(nums):
          if nums[j]!=0:
            nums[i] = nums[j]
            i += 1
          j += 1
        
        while i<len(nums):
          nums[i] = 0
          i += 1
        return

File: tnzhou/283.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nonZeroPoint = 0;
        for (int i = 0; i < nums.size(); i ++) {
            if (nums[i] != 0) 
                nums[nonZeroPoint++] = nums[i];
        }
        for (int i = nonZeroPoint; i < nums.size(); i ++) {
            nums[i] = 0;
        }
    }
};
